controller/iteration.go

```diff
@@ -213,10 +213,52 @@ func (c *IterationController) Update(ctx *app.UpdateIterationContext) error {
 		if ctx.Payload.Data.Attributes.UserActive != nil {
 			itr.UserActive = *ctx.Payload.Data.Attributes.UserActive
 		}
+		var oldSubtree []iteration.Iteration
+		if ctx.Payload.Data.Relationships != nil && ctx.Payload.Data.Relationships.Parent != nil {
+			// update parent of Iteration
+			// do not allow root-iteraiton to update its parent
+			if itr.IsRoot(itr.SpaceID) {
+				return jsonapi.JSONErrorResponse(ctx,
+					errors.NewForbiddenError("Parent of root iteration can not be updated"))
+			}
+			newParentID := ctx.Payload.Data.Relationships.Parent.Data.ID
+			if newParentID == nil {
+				return jsonapi.JSONErrorResponse(ctx,
+					errors.NewBadParameterError("Data.Relationships.Parent.ID", newParentID).Expected("not nil"))
+			}
+			id, err := uuid.FromString(*newParentID)
+			if err != nil {
```

Comment:
I guess we should also check the case in which you want to make the iteration a child of itself.

Reply:
yes done in https://github.com/fabric8-services/fabric8-wit/pull/1770/commits/accdb037340d51a5d0a9c5f852b382a8498846ef